examples/Example14/src/PhysicsList.cc

```diff
@@ -0,0 +1,291 @@
+#include "G4EmStandardPhysics_option4.hh"
+#include "G4EmStandardPhysicsWVI.hh"
+#include "G4EmStandardPhysicsGS.hh"
+#include "G4EmStandardPhysicsSS.hh"
+
+#include "G4EmLivermorePhysics.hh"
+#include "G4EmPenelopePhysics.hh"
+#include "G4EmLowEPPhysics.hh"
+
+#include "G4LossTableManager.hh"
+#include "G4UnitsTable.hh"
+#include "G4SystemOfUnits.hh"
+
+// particles
+
+#include "G4BosonConstructor.hh"
+#include "G4LeptonConstructor.hh"
+#include "G4MesonConstructor.hh"
+#include "G4BosonConstructor.hh"
+#include "G4BaryonConstructor.hh"
+#include "G4IonConstructor.hh"
+#include "G4ShortLivedConstructor.hh"
+
+
+// HepEm: the G4HepEm EM phyics constructor is (see PhysListHepEm)
+// G4Em : the G4 EM phyics constructor, that is equivalent to the G4HepEm, is (see PhysListG4Em)
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+PhysicsList::PhysicsList() : G4VModularPhysicsList(),
+ fEmPhysicsList(0), fStepMaxProcess(0), fMessenger(0)
+{
+  G4LossTableManager::Instance();
+  SetDefaultCutValue(1*mm);
+
+  fMessenger = new PhysicsListMessenger(this);
+  SetVerboseLevel(1);
+
+  // EM physics: set to HepEm by def.
+  fEmName        = G4String("HepEm");
+  fEmPhysicsList = new PhysListHepEm(fEmName);
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+PhysicsList::~PhysicsList()
+{
+  delete fMessenger;
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+void PhysicsList::ConstructParticle()
+{
+    G4BosonConstructor  pBosonConstructor;
+    pBosonConstructor.ConstructParticle();
+
+    G4LeptonConstructor pLeptonConstructor;
+    pLeptonConstructor.ConstructParticle();
+
+    G4MesonConstructor pMesonConstructor;
+    pMesonConstructor.ConstructParticle();
+
+    G4BaryonConstructor pBaryonConstructor;
+    pBaryonConstructor.ConstructParticle();
+
+    G4IonConstructor pIonConstructor;
+    pIonConstructor.ConstructParticle();
+
+    G4ShortLivedConstructor pShortLivedConstructor;
+    pShortLivedConstructor.ConstructParticle();
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+#include "G4ProcessManager.hh"
+
+void PhysicsList::ConstructProcess()
+{
+  // Transportation
+  //
+  AddTransportation();
+
+  // electromagnetic Physics List
+  //
+  fEmPhysicsList->ConstructProcess();
+  // add other processes if not HepEm physics list is used
+  if (fEmName!="HepEm" && fEmName!="G4Em" ) {
+    // decay Process
+    AddDecay();
+    // radioactive decay Process
+    AddRadioactiveDecay();
+  }
+
+  for(auto itr = (G4VMPLsubInstanceManager.offset[g4vmplInstanceID]).physicsVector->begin(); 
+  itr != (G4VMPLsubInstanceManager.offset[g4vmplInstanceID]).physicsVector->end();
+      ++itr)
+  {
+    (*itr)->ConstructProcess();
+  }
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+void PhysicsList::AddPhysicsList(const G4String& name)
```

Comment:
As said before, I don't think we need all this switching, we want G4HepEm on the host for comparison. However, I do think it would be critical to have an `FTFP_BERT` like physics list, including hadronic interactions, with only the EM physics for e-/e+/gamma replaced with G4HepEm or AdePT.

Reply:
This is done like this now.